pages/api/internal/assets/[anchor]/[assetType]/index.js

```diff
@@ -140,17 +146,31 @@ export default async function handle(req, res) {
 
 		if (existingAsset) {
 			await connectWithExistingAsset(existingAsset, nft)
-
+			rmSync(uploadedFile.path)
 			return res.json({
 				existingAsset
 			})
 		} else {
 			try {
-				writeFileSync(path.join(STORAGE_DIR, fileHash), fileContent)
+				// Split files per Smart contract. 
+				// This is a logical separation and also reduces the risk of Filesystem-overload
+				// due to too many files in the same directory
+				const storageFileName = path.join(nft.contract.csn, fileHash)
```

Comment:
This bit is confusing: There is a database field (and this one) called storageFile**Name**, but it's not a name but a relative path. I'd rename variables here, and also refactor computation of `targetDir` to avoid the `dirname` call -- you know the directory already.

Reply:
Removed https://github.com/authenticvision/digitalsoul/pull/41/commits/8807190ea224f95c7a949f5f541f4b70aea8b306